src/app/task.service.ts

```diff
@@ -21,14 +27,69 @@ export class TaskService {
   }
 
   deleteTask(id: number) {
-    this.taskList = this.taskList.filter(item => item.id !== id);
+    this.http.delete(`http://localhost:3001/removeTask/${id}`)
+      .pipe(
+        catchError((error) => {
+          console.error('Error deleting task:', error);
+          return [];
+        })
+      )
+      .subscribe({
+        next: () => {
+          this.taskList = this.taskList.filter(item => item.id !== id);
+        },
+        error: (error) => {
+          console.log(error);
+        }
+      });
   }
 
-  getTaskByID(id: string) {
-    return this.taskList.find(t => t.id == Number(id)) ?? null;
+  getTaskByID(id: string): Observable<TaskData | null> {
+    return this.http.get<TaskData>(`http://localhost:3001/task/${id}`)
+      .pipe(
+        catchError((error) => {
+          console.error('Error fetching task:', error);
+          return [null];
+        })
+      );
   }
 
   createTask(task: TaskData) {
-    this.taskList.push(task);
+    this.http.post(`http://localhost:3001/createTask`, task)
+      .pipe(
+        catchError((error) => {
+          console.error('Error creating task:', error);
+          return [];
+        })
+      )
+      .subscribe({
+        next: () => {
+          this.taskList = [...this.taskList, task];
+        },
+        error: (error) => {
+          console.log(error);
+        }
+      });
+  }
+
+  private loadTasksFromAPI() {
+    this.http.get<TaskList>("http://localhost:3001/allTasks/")
```

Comment:
also create a constants file or some sort of file where you will keep your constants, so you dont have to write http://localhost:3001 all the time and can jsut create a reusable variable for it, additionally you can set up some environments if you want, like evironment.ts and environment.dev.ts which would have different properties, of course based on the environment